src/db.py

```diff
@@ -0,0 +1,132 @@
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+DATABASE_NAME = 'instacart'
+SQLITE_DB_STRING = 'sqlite:///data/{}.db'.format(DATABASE_NAME)
+
+conn_type = 'mysql+pymysql'
+user = os.environ.get('MYSQL_USER')
+password = os.environ.get('MYSQL_PASSWORD')
+host = os.environ.get('MYSQL_HOST')
+port = os.environ.get('MYSQL_PORT')
+RDS_DB_STRING = '{}://{}:{}@{}:{}/{}'\
+                .format(conn_type, user, password, host, port, DATABASE_NAME)
+
+
+class OrderType(Base):
+    """Create a data model for order types derived from cluster centroids."""
+    __tablename__ = 'ordertypes'
+
+    label = Column(Integer, primary_key=True)
+
+    # Described by means
+    reordered = Column(Float, unique=False, nullable=False)
+    organic = Column(Float, unique=False, nullable=False)
+    popular = Column(Float, unique=False, nullable=False)
+    prepared = Column(Float, unique=False, nullable=False)
+    dairy = Column(Float, unique=False, nullable=False)
+    gluten = Column(Float, unique=False, nullable=False)
+    snack = Column(Float, unique=False, nullable=False)
+    meat = Column(Float, unique=False, nullable=False)
+    fish = Column(Float, unique=False, nullable=False)
+    beverage = Column(Float, unique=False, nullable=False)
+    veg = Column(Float, unique=False, nullable=False)
+    order_size = Column(Float, unique=False, nullable=False)
+
+    # Described by modes
+    order_dow = Column(Integer, unique=False, nullable=False)
+    order_hour_of_day = Column(Integer, unique=False, nullable=False)
+    days_since_prior_order = Column(Integer, unique=False, nullable=False)
+    order_size = Column(Integer, unique=False, nullable=False)
+
+    def __repr__(self):
+        return '<OrderType %s>' % self.label
+
+
+def run_ingest(engine_string, baskets_path):
+    orders = pd.read_csv(baskets_path)
+    amode = lambda x: mode(x)[0]
+
+    logger.debug('Aggregating orders by label')
+    order_types = orders.groupby('label').agg({
+        'reordered':    'mean',
+        'organic':      'mean',
+        'popular':      'mean',
+        'prepared':     'mean',
+        'dairy':        'mean',
+        'gluten':       'mean',
+        'snack':        'mean',
+        'meat':         'mean',
+        'fish':         'mean',
+        'beverage':     'mean',
+        'veg':          'mean',
+        'order_size':   'mean',
+
+        'order_dow':              amode,
+        'order_hour_of_day':      amode,
+        'days_since_prior_order': amode,
+        'order_size':             amode
+    })
+    del orders
+
+    logger.info('Connecting to: %s', engine_string)
+    engine = sql.create_engine(engine_string)
+
+    logger.info('Writing %d order types to database', len(order_types))
+    order_types.index = order_types.index.astype(int)
+    cols = ['order_dow', 'order_hour_of_day', 'days_since_prior_order', 'order_size']
+    order_types[cols] = order_types[cols].astype(int)
```

Comment:
how does the above hardcoding combine to looping through columns, checking for type, and converting to int if not int?